Lesson 06 postcss

Deman4ig opened the pull request.

Deman4ig committed:
Install postcss with plugins

Deman4ig committed:
Install postcss with plugins

Deman4ig committed:
Merge branch 'master' of github.com:js-tasks-ru/htmlcss-20210513_deman4ig into lesson-06_postcss

Deman4ig committed:
Add CSS reset and define project typography

Deman4ig committed:
Merge branch 'lesson-06_postcss' of github.com:Deman4ig/htmlcss-20210513_deman4ig into lesson-06_postcss

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

Deman4ig committed:
Install and configure stylelint

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov

Deman4ig committed:
Install and configure prettier

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

GSemikozov commented on assets/fonts/typography.scss:
```diff
@@ -0,0 +1,68 @@
+$color: var(--black, #212429);
```
для переменных можно создать отдельный файл variables.scss и уже оттуда забирать. Сам файл можно поместить например в `assets/css/config/`

GSemikozov commented on assets/fonts/typography.scss:
```diff
@@ -0,0 +1,68 @@
+$fontFamily: 'Inter, sans-serif';
+$mobileBreakpoint: 767px;
+
+@mixin typography {
```
и для миксинов, если они переиспользуются, тоже можно создать отдельный файл

Deman4ig replied:
сделал!

GSemikozov commented on assets/fonts/typography.scss:
```diff
@@ -0,0 +1,68 @@
+    font-size: 50px;
+    line-height: 64px;
+
+    @media screen and (max-width: $mobileBreakpoint) {
```
для медиа-запросов было бы здорово добавить свой миксин

Deman4ig replied:
разобрался! спасибо за подсказку!

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

Deman4ig committed:
Refactor typography.scss

1. Add media query mixin
2. Move mixins to a separate file
3. Rename build script output to style.css

jsru commented:
Решение было обновлено, посмотрим что скажет @GSemikozov
